pkg/services/authn/authnimpl/sync/rbac_sync.go

```diff
@@ -34,19 +35,57 @@ func (s *RBACSync) SyncPermissionsHook(ctx context.Context, ident *authn.Identit
 		return nil
 	}
 
-	permissions, err := s.ac.GetUserPermissions(ctx, ident, accesscontrol.Options{ReloadCache: false})
+	// Populate permissions from roles
+	permissions, err := s.fetchPermissions(ctx, ident)
 	if err != nil {
-		s.log.FromContext(ctx).Error("Failed to fetch permissions from db", "error", err, "id", ident.ID)
-		return errSyncPermissionsForbidden
+		return err
 	}
 
 	if ident.Permissions == nil {
-		ident.Permissions = make(map[int64]map[string][]string)
+		ident.Permissions = make(map[int64]map[string][]string, len(permissions))
```

Comment:
I don't think the `len(permissions)` capacity here is what you want. That sets the capacity of the `map[int64]` which is the org IDs. Since you know you're setting only one org in this function maybe just using the value `1` is ok.